### ISSUE TITLE
Needing crypto_core_ristretto255_{from,to}bytes()

### ISSUE THREAD

**Issue by fhajji** [jedisct1/libsodium#928]
I have an application that requires mapping short (32 bytes, I don't care about the sign bit) messages onto Points of the Ristretto group, perform a lot of calculations on that group, and then map the resulting Points back to messages.

The code would basically go like this:
```c
void message_to_point(unsigned char *point, const unsigned char *message) {
  ge25519_p3 h;
  int rc = ristretto255_frombytes(&h, message);
  ge25519_p3_tobytes(point, &h);
}
```
and
```c
void point_to_message(unsigned char *message, const unsigned char *point) {
  ge25519_p3 h;
  int rc = ge25519_frombytes(&h, point);
  ristretto255_p3_tobytes(message, &h);
}
```
Unfortunately, nothing of this is user-accessible. Currently, I need to manually add src/sodium/crypto_core/ed25519/ref10/ed25519_ref10.c and associated files to my project, which is very inconvenient.

Oh, and before someone asks why I don't use ristretto255_elligator() or ristretto255_from_hash(): the latter is not invertible, and the former is not applicable, because I do perform calculations on the resulting Points, which may result in Ristretto Points that have no pre-images under Elligator. Furthermore, since the messages are not uniformly distributed, Elligator would be pointless anyway. Furthermore, the messages are never send in plain text over the wire.

So, to sum up: it would be really great to have something like crypto_core_ristretto255_frombytes() and crypto_core_ristretto255_tobytes(). Of course, with the corresponding caveats in the documentation.

**Comment by jedisct1** [comment]
This is not planned, sorry.

**Comment by fhajji** [comment]
Okay, no problem (forking). Thanks and keep up the good work.
